src/core/Flora/Import/Package.hs

```diff
@@ -227,37 +228,43 @@ loadAndExtractCabalFile userId filePath = loadFile filePath >>= extractPackageDa
 {-| Persists an 'ImportOutput' to the database. An 'ImportOutput' can be obtained
  by extracting relevant information from a Cabal file using 'extractPackageDataFromCabal'
 -}
-persistImportOutput :: (DB :> es, IOE :> es) => ImportOutput -> Eff es ()
-persistImportOutput (ImportOutput package categories release components) = do
-  pool <- getPool
+persistImportOutput :: (DB :> es, IOE :> es) => Poolboy.WorkQueue -> ImportOutput -> Eff es ()
+persistImportOutput wq (ImportOutput package categories release components) = do
+  dbPool <- getPool
   liftIO . T.putStrLn $! "📦  Persisting package: " <> packageName <> ", 🗓  Release v" <> display (release.version)
   persistPackage
   Update.upsertRelease release
-  parallelRun pool (persistComponent pool) components
+  parallelRun dbPool (persistComponent dbPool) components
   liftIO $! putStr "\n"
   where
     parallelRun :: (MonadIO m, Foldable t) => Pool Connection -> (a -> Eff [DB, IOE] b) -> t a -> m ()
-    parallelRun pool f = force $! liftIO . S.drain . S.fromParallel . S.mapM (runEff . runDB pool . f) . S.fromFoldable
+    parallelRun pool f xs = liftIO $ forM_ xs $ Poolboy.enqueue wq . void . runEff . runDB pool . f
     packageName = display (package.namespace) <> "/" <> display (package.name)
     persistPackage = do
       let packageId = package.packageId
       Update.upsertPackage package
       forM_ categories (\case Tuning.NormalisedPackageCategory cat -> Update.addToCategoryByName packageId cat)
 
-    persistComponent pool (packageComponent, deps) = do
+    persistComponent dbPool (packageComponent, deps) = do
       liftIO . T.putStrLn $
         "🧩  Persisting component: "
           <> display (packageComponent.canonicalForm)
           <> " with "
           <> display (length deps)
           <> " dependencies."
       Update.upsertPackageComponent packageComponent
-      parallelRun pool persistImportDependency deps
+      parallelRun dbPool persistImportDependency deps
 
     persistImportDependency dep = do
       Update.upsertPackage (dep.package)
       Update.upsertRequirement (dep.requirement)
 
+withWorkerDbPool :: (IOE :> es) => (Poolboy.WorkQueue -> Eff es a) -> Eff es a
+withWorkerDbPool f =
+  withEffToIO $ \effIO ->
+    Poolboy.withPoolboy (Poolboy.poolboySettingsWith 100) $ \wq ->
```

Comment:
Hard-coded, but I  adjust it to the number of the connections pool if you pre-accept my PR.